File: FootballExplorer_java/app/src/main/java/fr/a42consulting/footballexplorer/View/ITeamsSearchView.java
package fr.a42consulting.footballexplorer.View;

import java.util.ArrayList;

import fr.a42consulting.footballexplorer.Model.TeamsSearchTeam;

public interface ITeamsSearchView {
    void onSearchResult(ArrayList<TeamsSearchTeam> teams);
    void onShowTeam(String team);
    void onMessage(String message);
}

File: FootballExplorer_kotlin/app/src/main/java/fr/a42consulting/footballexplorer/view/ITeamsSearchView.kt
package fr.a42consulting.footballexplorer.view

import java.util.ArrayList

import fr.a42consulting.footballexplorer.model.TeamsSearchTeam

interface ITeamsSearchView {
    fun onSearchResult(teams: ArrayList<TeamsSearchTeam>, isNewSearch: Boolean)
    fun onShowTeam(team: String)
    fun onMessage(message: String)
}
